### PR TITLE
Adjust Serverless tests to the new Function CR

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Adjusted serverless tests
- Little code cleanup
- Enabled Function Controller by default

**Related issue(s)**

#7359

### CODE REVIEW

File: resources/function-controller/values.yaml [hunk 1]
```diff
@@ -8,15 +8,19 @@ tests:
...
+    requests:
+      memory: 32Mi
+      cpu: 100m
     limits:
-      memory: 128Mi
+      memory: 64Mi
+      cpu: 200m
   envs:
-    waitTimeout: 45m # sorry guys :sadgabens:
+    waitTimeout: 5m
```
Comment (aerfio) [inline resources/function-controller/values.yaml:23]: I would not be so brave here 😛

File: tests/function-controller/pkg/resource/resource.go [hunk 2]
```diff
@@ -17,20 +16,24 @@ import (
 	watchtools "k8s.io/client-go/tools/watch"
 )
 
+type logger interface {
+	Logf(format string, args ...interface{})
+}
+
```
Comment (aerfio) [inline tests/function-controller/pkg/resource/resource.go:22]: you've already defined this interface in apirule.go - move this to some kind of shared place

File: tests/function-controller/pkg/retry/retry.go [hunk 3]
```diff
@@ -15,7 +13,11 @@ var (
 	ErrInvalidFunc = goerrors.New("invalid function")
 )
 
-func fnWithIgnore(fn func() error, ignoreErr func(error) bool, callbacks ...func(...interface{})) func() error {
+type logger interface {
+	Logf(format string, args ...interface{})
+}
```
Comment (aerfio) [inline tests/function-controller/pkg/retry/retry.go:18]: again, same as before

File: tests/function-controller/testsuite/function.go [hunk 4]
```diff
@@ -2,101 +2,96 @@ package testsuite
...
 )
 
 var (
 	ErrInvalidDataType = errors.New("invalid data type")
 )
 
-const ready = "Running"
+type logger interface {
+	Logf(format string, args ...interface{})
+}
```
Comment (aerfio) [inline tests/function-controller/testsuite/function.go:27]: .

File: tests/function-controller/testsuite/testsuite.go [hunk 5]
```diff
@@ -52,9 +52,11 @@ func New(restConfig *rest.Config, cfg Config, t *testing.T, g *gomega.GomegaWith
 		return nil, errors.Wrap(err, "while creating K8s Dynamic client")
 	}
 
-	ns := namespace.New(coreCli, cfg.Namespace)
-	f := newFunction(dynamicCli, cfg.FunctionName, cfg.Namespace, cfg.WaitTimeout, t.Logf)
-	ar := apirule.New(dynamicCli, cfg.APIRuleName, cfg.Namespace, cfg.WaitTimeout, t.Logf)
+	namespaceName := fmt.Sprintf("%s-%d", cfg.Namespace, rand.Uint32())
```
Comment (aerfio) [inline tests/function-controller/testsuite/testsuite.go:55]: oh, is that a reason you've deleted first cleanup? new namespace + hostname every time?

File: tests/function-controller/pkg/apirule/apirule.go [hunk 6]
```diff
@@ -12,27 +12,33 @@ import (
 	"k8s.io/client-go/dynamic"
 )
 
+type logger interface {
+	Logf(format string, args ...interface{})
+}
+
```
Comment (aerfio) [inline tests/function-controller/pkg/apirule/apirule.go:18]: .

File: resources/function-controller/values.yaml (hunk 1 above)
Comment (michal-hudy) [reply resources/function-controller/values.yaml:23]: As you probably noticed, it passed on 4 pipelines :P For minikube there will proper override.

File: resources/function-controller/values.yaml [hunk 7]
```diff
@@ -20,7 +20,7 @@ tests:
       memory: 64Mi
       cpu: 200m
   envs:
-    waitTimeout: 5m
+    waitTimeout: 10m
```
Comment (aerfio) [inline resources/function-controller/values.yaml:23]: 😀

**Pull request merged**